src/Locks.ts

```diff
@@ -0,0 +1,249 @@
+import { Types } from 'ably';
+
+import Space from './Space.js';
+import type { SpaceMember } from './types.js';
+import type { PresenceMember } from './utilities/types.js';
+import { ERR_LOCK_IS_LOCKED, ERR_LOCK_INVALIDATED, ERR_LOCK_REQUEST_EXISTS, ERR_LOCK_RELEASED } from './Errors.js';
+import EventEmitter, {
+  InvalidArgumentError,
+  inspect,
+  type EventKey,
+  type EventListener,
+} from './utilities/EventEmitter.js';
+
+export type LockStatus = 'pending' | 'locked' | 'unlocked';
+
+export type Lock = {
+  member: SpaceMember;
+  request: LockRequest;
+};
+
+export type LockRequest = {
+  id: string;
+  status: LockStatus;
+  timestamp: number;
+  attributes?: LockAttributes;
+  reason?: Types.ErrorInfo;
+};
```

Comment:
it would be useful to developers to have these types `src/types.d.ts` so they can use it in their code

Reply:
Do you mean move all types defined in `Locks.js` into `types.d.ts`?

Reply:
The types defined above, which I think developers will interact with `LockStatus`, `Lock` and `LockRequest`

Reply:
Ok got it, I exported them in `index.ts`, which I assume is enough? See 77d2dc8.

Reply:
As discussed in stand-up, moved to `types.d.ts` in 86712a5.